llvm/tools/llvm-jitlink/llvm-jitlink-elf.cpp

```diff
@@ -194,7 +194,8 @@ Error registerELFGraphInfo(Session &S, LinkGraph &G) {
     else
       FileInfo.SectionInfos[Sec.getName()] = {
           ArrayRef<char>(FirstSym->getBlock().getContent().data(), SecSize),
-          SecAddr.getValue(), FirstSym->getTargetFlags()};
+          (SecAddr - FirstSym->getOffset()).getValue(),
```

Comment:
Instead of setting this here, it seems like it'd be tidier to define SecAddr like this on line 183:
```c++
auto SecAddr = FirstSym->getAddress() - FirstSym->getOffset();
```
Unless you've already tried that and hit other issues I'm happy to go ahead and make that change.